src/coreclr/jit/fgopt.cpp

```diff
@@ -1401,12 +1401,6 @@ bool Compiler::fgOptimizeEmptyBlock(BasicBlock* block)
                 break;
             }
 
-            // can't allow fall through into cold code
-            if (block->IsLastHotBlock(this))
```

Comment:
This check looks like it wouldn't pessimize anything if we aren't hot/cold splitting, but `IsLastHotBlock` returns true if `block == fgLastBB` too, so this check was blocking empty block removal for the last block in the worklist. Thus, the check's removal incurs [some small diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=887522&view=ms.vss-build-web.run-extensions-tab). PerfScore diffs look wonky due to churn in `optSetBlockWeights`, but the actual layout diffs look like an improvement.